File: golang/lib/kurtosis-lambda/kurtosis_lambda.go
/*
 * Copyright (c) 2020 - present Kurtosis Technologies LLC.
 * All Rights Reserved.
 */

package kurtosis_lambda

import "github.com/kurtosis-tech/kurtosis-client/golang/lib/networks"

type KurtosisLambda interface {
	Execute(networkCtx *networks.NetworkContext, serializedParams string) (serializedResult string, resultError error)
}

File: typescript/src/lib/kurtosis_lambda/kurtosis_lambda.ts
import { NetworkContext } from "kurtosis-core-api-lib";
import { Result } from "neverthrow";

// Docs available at https://docs.kurtosistech.com/kurtosis-lambda-api-lib/lib-documentation
export interface KurtosisLambda {
	execute(
        networkCtx: NetworkContext, 
        serializedParams: string
    ): Promise<Result<string, Error>>
}